test/accessor_tests.cc

```diff
@@ -654,29 +654,34 @@ namespace detail {
 #if !CELERITY_ACCESSOR_BOUNDARY_CHECK
 		SKIP("CELERITY_ACCESSOR_BOUNDARY_CHECK=0");
 #endif
-
 		buffer<int, Dims> buff(range_cast<Dims>(range<3>{10, 20, 30}));
 		const auto accessible_sr = subrange_cast<Dims>(subrange<3>{{5, 10, 15}, {1, 2, 3}});
 		const auto oob_idx_lo = id_cast<Dims>(id<3>{1, 2, 3});
 		const auto oob_idx_hi = id_cast<Dims>(id<3>{7, 13, 25});
-		distr_queue q;
 
-		celerity::test_utils::log_capture lc;
+		// we need to be careful about the orderign of the construction and destruction
+		// of the Celerity queue and the log capturing utility here
+		std::unique_ptr<celerity::test_utils::log_capture> lc;
```

Comment:
Maybe it's because my brain is pretty fried atm but from the patch alone I can't really tell what the issue is. Could you elaborate? Also I found [this](https://github.com/celerity/celerity-runtime/blob/79f97c2b2b5b1442fec2668cfbf813aa50cb0225/test/system/distr_tests.cc#L140) - possibly related?

Reply:
In debug, messages are printed to the log from another thread(s), so we need to ensure that the queue is dead before we read the capture.

Reply:
Can't we just switch to a thread-safe `spdlog::sinks::ostream_sink_mt` then?

Reply:
I just tested this, and it does not fix the segfaults.
I believe this is because, while the sink is MT safe, adding and removing sinks might not be. But this is just an educated guess. This seems likely though, because ultimately "sinks" is just a std vector. Since we have no way to control how it is accessed within the library (at least that I can think of) I believe we need to resort to controlling the lifetimes.